controllers/category.controller.js

```diff
@@ -0,0 +1,83 @@
+const db = require('../DB/connection');
+const categoryDecorator = require('../decorators/category.decorator')
+
+exports.getAll = async (req, res) => {
+    try {
+        const userId = 1
+        // const userId = req.user.id;
+        const [categories] = await db.query('SELECT * FROM categories WHERE user_id = ?', [userId]);
+        res.json({ data: categories.map(categoryDecorator) });
+    } catch (error) {
+        res.json(error);
+    }
+};
+
+exports.getOne = async (req, res) => {
+    try {
+        const userId = 1
+        // const userId = req.user.id;
+        if (isNaN(req.params.id)) {
+            return res.status(422).json({ message: 'The parameter must be a number' });
+        }
+        const [category] = await db.query('SELECT * FROM categories WHERE id = ? AND user_id =?', [req.params.id, userId]);
+        if (category.length === 0) {
+            return res.status(404).json({ message: "Category not found" });
+        }
+        res.json({ data: category.map(categoryDecorator) });
+    } catch (error) {
+        res.json(error);
+    }
+};
+
+exports.create = async (req, res) => {
+    const { name } = req.body;
+    try {
+        const userId = 1
+        // const userId = req.user.id;
+        const [result] = await db.query('INSERT INTO categories (name, user_id) VALUES (?,?) ', [name, userId]);
+        const newCategory = { id: result.insertId, name }
+        res.status(201).json({ data: categoryDecorator(newCategory) });
+    } catch (error) {
+        res.json({data:error});
+    }
+};
+
+exports.update = async (req, res) => {
+    const { name } = req.body;
+    const { id } = req.params;
+
```

Comment:
Get Category by id, before perform the change

Reply:
ON the first time get the category by id, then confirm we are applying any change, if not just return the category as it is. If so update the category.